test/functional/assumevalid.py

```diff
@@ -166,25 +186,19 @@ def run_test(self):
         node1.send_header_for_blocks(self.blocks[2000:])
         node2.send_header_for_blocks(self.blocks[0:200])
 
-        # Send 102 blocks to node0. Block 102 will be rejected.
-        for i in range(101):
-            node0.send_message(msg_block(self.blocks[i]))
-        node0.sync_with_ping() # make sure the most recent block is synced
-        node0.send_message(msg_block(self.blocks[101]))
-        assert_equal(self.nodes[0].getblock(self.nodes[0].getbestblockhash())['height'], 101)
+        # Send blocks to node0. Block 102 will be rejected.
+        self.send_blocks_until_disconnected(node0)
+        self.assert_blockchain_height(self.nodes[0], 101)
 
-        # Send 3102 blocks to node1. All blocks will be accepted.
+        # Send all blocks to node1. All blocks will be accepted.
         for i in range(2202):
             node1.send_message(msg_block(self.blocks[i]))
-        node1.sync_with_ping() # make sure the most recent block is synced
+        node1.sync_with_ping()  # make sure the most recent block is synced
```

Comment:
needs https://github.com/bitcoin/bitcoin/pull/10114 as mentioned in the original PR

Reply:
https://github.com/dashpay/dash/pull/2771/commits/b0b1bbe90b8bff12265fe9dc16d0017fa09a2b34